pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -75,13 +75,7 @@ impl InMemoryLayerInner {
     }
 }
 
-impl InMemoryLayer {
-    pub fn get_timeline_id(&self) -> TimelineId {
-        self.timeline_id
-    }
-}
-
-impl Layer for InMemoryLayer {
+impl LayerRange for InMemoryLayer {
```

Comment:
Is this desirable? The `InMemoryLayer` has a variable lsn range, and someone might expect `get_lsn_range` to be constant, similarly to how `hash()` should be constant for anything that goes in a hashmap. See discussion here https://github.com/neondatabase/neon/pull/2998#discussion_r1080605726 Definitely the new layer map makes such an assumption, and if `LayerRange` is the only requirement for a layer going into the layer map, then it should be required that the range remains fixed.

Maybe just return the full key range (conservative) in `get_lsn_range`, and if necessary add a separate function to narrow the range down. Or don't implement this trait for the InMemoryLayer if it's not needed.